tests/lib/rules/no-unused-prop-types.js

```diff
@@ -784,54 +764,6 @@ ruleTester.run('no-unused-prop-types', rule, {
       ].join('\n')
     }, {
       code: [
-        'const statelessComponent = (props) => {',
```

Comment:
I think that it ideally should support this use case, but our mechanism of detecting components may make it difficult.